### PR TITLE
Add auto-generate sample for ProviderHub and Resources

### PR DESCRIPTION
**Pull request by dvbb**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/providerhub/Azure.ResourceManager.ProviderHub/src/autorest.md [hunk 1]
```diff
@@ -49,6 +49,12 @@ rename-rules:
   URI: Uri
   Etag: ETag|etag
 
+request-path-to-resource-name:
+  /subscriptions/{subscriptionId}/providers/Microsoft.ProviderHub/providerRegistrations/{providerNamespace}/resourcetypeRegistrations/{resourceType}/skus/{sku}: NestedResourceTypeSku
```
Comment (fengzhou-msft) [inline sdk/providerhub/Azure.ResourceManager.ProviderHub/src/autorest.md:53]: Should this be `ResourceTypeSku`?

Comment (dvbb) [reply sdk/providerhub/Azure.ResourceManager.ProviderHub/src/autorest.md:53]: It already has a model called [ResourceTypeSku](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/providerhub/Azure.ResourceManager.ProviderHub/src/Generated/Models/ResourceTypeSku.cs#L16)

Comment (fengzhou-msft) [reply sdk/providerhub/Azure.ResourceManager.ProviderHub/src/autorest.md:53]: Can we rename that model to something like `ResourceTypeSkuInfo`? I don't think `{providerNamespace}/resourcetypeRegistrations/{resourceType}` is considered a nested resource. We should count the levels starting from `{providerNamespace}`. Therefore, this is a top level resource after it's registered altough we're registering it under `Microsoft.ProviderHub/providerRegistrations`. And the description of the `CreateOrUpdate` method of this request path also says `Creates or updates the resource type skus in the given resource type.`.

Comment (dvbb) [reply sdk/providerhub/Azure.ResourceManager.ProviderHub/src/autorest.md]: fixed, change `NestedResourceTypeSku` => `ResourceTypeSkuInfo`

Comment (fengzhou-msft) [reply sdk/providerhub/Azure.ResourceManager.ProviderHub/src/autorest.md]: I mean we use `ResourceTypeSku` for the resource name and rename the existing  `ResourceTypeSku` model to `ResourceTypeSkuInfo`. This way the resource names are relatively consistent.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**